compiler/rustc_lexer/src/unescape.rs

```diff
@@ -351,7 +338,7 @@ where
     }
 }
 
-fn byte_from_char(c: char) -> u8 {
+pub fn byte_from_char(c: char) -> u8 {
```

Comment:
This one is a bit dicey. I agree that in the context of compiler we don't really care about nice abstrction boundaries, and this is a simplification. 

But we want `rustc_lexer` to also be a nice crates.io crates, and from that point of view this feels a bit too type-unsafe. 

No strong opinion overall. 

Though, if we do this, this one surely wants to be tagged as `#[inline]`? tiny cross-crate fn.

Reply:
It's a fair point. A couple of questions:
- Is `rustc_lexer` actually usable outside of rustc/rust-analyzer? the `rustc-lexer` crate is v0.1.0 and hasn't been updated in three years. I see some auto-updated variants but none of them have been updated in the past year.
- I'm not sure how this is type-unsafe. You'll get `u8`/`char` type mismatches if you do something wrong?
- Inlining is a good idea.